### PR TITLE
refact drf plugin ut

### LINKED ISSUE #2810: Part of the ut shows that the operation is successful, but the test case is not actually executed
**What happened**:
In [#2592](https://github.com/volcano-sh/volcano/pull/2592), modify the construction method of some UT use cases cache.schedulerCache, construct config through `kube.BuildConfig` and generate the corresponding SchedulerCache, there are the following two problems:
1. When the UT use case is running, the environment does not have k8s related information and configuration, and it is not particularly reasonable to construct config through kube.BuildConfig
2. After kube.BuildConfig returns error, return directly without adding t.Errorf() information. The test framework thinks that ut runs successfully, but it does not actually run

**The test cases involved are:**

1. volcano.sh/volcano/pkg/scheduler/actions/allocate
2. volcano.sh/volcano/pkg/scheduler/actions/preempt
3. volcano.sh/volcano/pkg/scheduler/actions/reclaim
4. volcano.sh/volcano/pkg/scheduler/actions/shuffle
5. volcano.sh/volcano/pkg/scheduler/plugins/binpack
6. volcano.sh/volcano/pkg/scheduler/plugins/drf
7. volcano.sh/volcano/pkg/scheduler/plugins/predicates
8. volcano.sh/volcano/pkg/scheduler/plugins/proportion
9. volcano.sh/volcano/pkg/scheduler/plugins/tdm

similar to the following：
https://github.com/volcano-sh/volcano/blob/d4458ffe81207749f04727873d37544d7138152d/pkg/scheduler/plugins/proportion/proportion_test.go#L205-L217
**What you expected to happen**:
The test case can run normally, the line coverage is not 0.

**How to reproduce it (as minimally and precisely as possible)**:
There are two ways to reproduce:
1. Add log `t.Errorf("kube build config error, err: %v", err)` where `config, err := kube.BuildConfig(option.KubeClientOptions)` reports an error return, execute `make unit-test` can find use cases that fail to run.
2. Execute `go test -v --covermode=count -coverprofile=count.out` in each ut test file directory to find ut use case files with 0 test coverage

**Anything else we need to know?**:

**Environment**:
- Volcano Version: master
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by lowang-bh**
Fixes #2810 

Some UT actually does not run.

image

### CODE REVIEW

File: pkg/scheduler/plugins/drf/hdrf_test.go [hunk 1]
```diff
@@ -49,82 +52,42 @@ type pgSpec struct {
...
-				},
+			TestCommonStruct: uthelper.TestCommonStruct{
+				Name:    "rescaling test",
+				Plugins: plugins,
+				PodGroups: []*schedulingv1.PodGroup{
+					util.BuildPodGroup("pg1", "default", "root-sci", 0, nil, schedulingv1.PodGroupInqueue),
+					util.BuildPodGroup("pg21", "default", "root-eng-dev", 0, nil, schedulingv1.PodGroupInqueue),
+					util.BuildPodGroup("pg22", "default", "root-eng-prod", 0, nil, schedulingv1.PodGroupInqueue),
+				},
+				Pods: mergePods(
```
Comment (Monokaix) [inline pkg/scheduler/plugins/drf/hdrf_test.go:71]: Is this method truly needed? We can declare a slice directly.

Comment (lowang-bh) [reply pkg/scheduler/plugins/drf/hdrf_test.go:71]: How？each makePods returns a list, and I need to merge them to one list. Append looks more complex

Comment (Monokaix) [reply pkg/scheduler/plugins/drf/hdrf_test.go:71]: OK

**Review by william-wang — APPROVED** [review]
/approve

**Pull request merged**